Classes/Format/Mods.php

```diff
@@ -78,56 +119,216 @@ public function extractMetadata(\SimpleXMLElement $xml, array &$metadata)
             }
         }
         // Get "year" and "year_sorting" if not specified separately.
-        $years = $xml->xpath('./mods:originInfo[not(./mods:edition="[Electronic ed.]")]/mods:dateIssued[@keyDate="yes"]');
+        $years = $this->xml->xpath('./mods:originInfo[not(./mods:edition="[Electronic ed.]")]/mods:dateIssued[@keyDate="yes"]');
         // Get "year" and "year_sorting" again if that was to sophisticated.
         if (empty($years)) {
             // Get all dates and assume these are dates of publication.
-            $years = $xml->xpath('./mods:originInfo/mods:dateIssued');
+            $years = $this->xml->xpath('./mods:originInfo/mods:dateIssued');
         }
         if (!empty($years)) {
             foreach ($years as $year) {
-                $metadata['year'][] = (string) $year;
-                if (!$metadata['year_sorting'][0]) {
+                $this->metadata['year'][] = (string) $year;
+                if (!$this->metadata['year_sorting'][0]) {
                     $year_sorting = str_ireplace('x', '5', preg_replace('/[^\d.x]/i', '', (string) $year));
                     if (
                         strpos($year_sorting, '.')
                         || strlen($year_sorting) < 3
                     ) {
                         $year_sorting = ((intval(trim($year_sorting, '.')) - 1) * 100) + 50;
                     }
-                    $metadata['year_sorting'][0] = intval($year_sorting);
+                    $this->metadata['year_sorting'][0] = intval($year_sorting);
                 }
             }
         }
     }
+
+    /**
+     * Get "description" stored in recordInfoNote.
+     *
+     * @access private
+     *
+     * @return void
+     */
+    private function getDescription() {
+        $this->getSingleMetadata('description', './mods:recordInfo/mods:recordInfoNote/text()');
+    }
+
+    /**
+     * Get "identifier" - example: hotels (built public accommodations), AAT ID: 300007166.
+     *
+     * @access private
+     *
+     * @return void
+     */
+    private function getIdentifier() {
+        $this->getSingleMetadata('identifier', './mods:identifier/text()');
+    }
+
+    /**
+     * Get "license" - license on which object is published.
+     *
+     * @access private
+     *
+     * @return void
+     */
+    private function getLicense() {
+        $this->getSingleMetadata('license', './mods:accessCondition/text()');
+    }
+
+    /**
+     * Get names of the original object:
+     *   - main name
+     *   - alternative names
+     *
+     * @access private
+     *
+     * @return void
+     */
+    private function getObjectNames() {
+        $this->getSingleMetadata('object_name', './mods:relatedItem/mods:titleInfo[not(@displayLabel="alternative")]/mods:title/text()');
+        $this->getSingleMetadata('object_alternative_names', './mods:relatedItem/mods:titleInfo[@displayLabel="alternative"]/mods:title/text()');
+    }
```

Comment:
See above: please add this as default mappings to `Resources/Private/Data/MetadataDefaults.php`.